### PR TITLE
Fix/user bubble/slot size

### PR DESCRIPTION
**Pull request by skjnldsv**
image

- Cleaning up docs
- Added new slot to add whatever we want inside the bubble after the label
- Added docs for this new feature too
- Removed the use of the Popover component if the popover was empty, this avoid a weird unecessary complex html structure for such a simple user bubble
- Added a way to set size and avatar margin for the whole component

image

### CODE REVIEW

**Comment by skjnldsv** [comment]
>  I have no idea what I'm reviewing here, sorry

See the vue docs deploy preview :)
I also added the list of changes :rocket: 

Basically, better perfs, improved simpler design, added size props & title slot

**Comment by violoncelloCH** [comment]
I can take a look this evening :)

File: src/components/UserBubble/UserBubble.vue [hunk 1]
```diff
@@ -152,33 +240,37 @@ export default {
...
+	}
 
-.user-bubble-primary {
-	color: var(--color-primary-text);
-	background-color: var(--color-primary-element);
-}
+	&__content {
+		display: inline-flex;
+		max-width: 100%;
+		background-color: var(--color-background-dark);
```
Comment (violoncelloCH) [inline src/components/UserBubble/UserBubble.vue:256]: Now that we have the title slot we should probably have `padding-right` on the content instead of margin-right on the title, as otherwise there is some issue at the right side of the bubble if there is something in the slot:
image

File: styleguide.config.js [hunk 2]
```diff
@@ -22,7 +22,7 @@ module.exports = {
 	components: 'src/components/*/*.vue',
 	getComponentPathLine(componentPath) {
 		const name = path.basename(componentPath, '.vue')
-		return `import { ${name} } from '@nextcloud/vue/dist/Components/${name}'`
+		return `import ${name} from '@nextcloud/vue/dist/Components/${name}'`
```
Comment (violoncelloCH) [inline styleguide.config.js:25]: is this related to the UserBubble component?

File: src/components/UserBubble/UserBubble.vue [hunk 3]
```diff
@@ -83,47 +106,100 @@ export default {
...
+		/**
+		 * Override generated avatar, can be an url or an icon class
+		 */
 		avatarImage: {
 			type: String,
-			default: '',
+			default: undefined,
+			validator: str => {
+
+			}
```
[the remark this answers is not in the document]
Comment (skjnldsv) [reply src/components/UserBubble/UserBubble.vue:117]: Nothing, leftover, sorry! Forgot to push :see_no_evil:

File: src/components/UserBubble/UserBubble.vue [hunk 4]
```diff
@@ -83,47 +106,100 @@ export default {
...
-			return false
+		/**
+		 * If userbubble is empty, let's NOT
+		 * use the Popovercomponent
+		 * @returns {boolean}
+		 */
+		isPopover() {
+			return !this.popoverEmpty
+				? 'Popover'
+				: 'div'
```
[the remark this answers is not in the document]
Comment (skjnldsv) [reply src/components/UserBubble/UserBubble.vue:179]: Definitely wrong! Thanks for the catch!

File: src/components/UserBubble/UserBubble.vue (hunk 1 above)
Comment (skjnldsv) [reply src/components/UserBubble/UserBubble.vue:256]: Good idea!!

File: styleguide.config.js (hunk 2 above)
Comment (skjnldsv) [reply styleguide.config.js:25]: Yes and no!
This is related to all components, I noticed the import text was wrong, we cannot import with the brackets if we use the direct component path.
I will move this to another pr for better clarity, let's not use this as a free pass!

**Comment by skjnldsv** [comment]
@violoncelloCH everything should be addressed now :)
I specifically left out the title slot from the auto ellipsis. Because it's up to the dev to put whatever and handle it. If it's plain text, considering the title attribute is most likely the bets option here :wink:
